docs/src/user_docs/assembly/cryptographic_operations.md

```diff
@@ -98,7 +98,7 @@ The following instructions are designed mainly for use in recursive verification
 | Instruction                         | Stack_input                                                                                       | Stack_output                                                                                        | Notes                                                                                                                                                                                                                                                                                                                          |
 | ----------------------------------- | ------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------- |--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | eval_circuit <br /> - *(1 cycle)*     | [ptr, n_read, n_eval, ...]                                                                        | [ptr, n_read, n_eval, ...]                                                                          | Evaluates an arithmetic circuit, and checks that its output is equal to zero. `ptr` specifies the memory address at which the circuit description is stored with the number of input extension field elements specified by `n_read` and the number of evaluation gates, encoded as base field elements, specified by `n_eval`. |
-| horner_eval_base <br /> - *(1 cycle)* | [c7,  c6,  c5,  c4,  c3,  c2,  c1,  c0, - , - , - , - , - , alpha_addr, acc1, acc0, ...]          | [c7,  c6,  c5,  c4,  c3,  c2,  c1,  c0, - , - , - , - , - , alpha_addr, acc1', acc0', ...]          | Performs 8 steps of the Horner evaluation method on a polynomial with coefficients over the base field, i.e., it computes `acc' = (((acc_tmp * alpha + c4) * alpha + c5) * alpha + c6) * alpha + c7` where `acc_tmp := (((acc * alpha + c0) * alpha + c1) * alpha + c2) * alpha + c3`                                          |
+| horner_eval_base <br /> - *(1 cycle)* | [c7,  c6,  c5,  c4,  c3,  c2,  c1,  c0, - , - , - , - , - , alpha_addr, acc1, acc0, ...]          | [c7,  c6,  c5,  c4,  c3,  c2,  c1,  c0, - , - , - , - , - , alpha_addr, acc1', acc0', ...]          | Performs 8 steps of the Horner evaluation method using 3-level computation: `tmp0 = ((acc * alpha + c0) * alpha) + c1`, `tmp1 = (((tmp0 * alpha + c2) * alpha + c3) * alpha) + c4`, `acc' = (((tmp1 * alpha + c5) * alpha + c6) * alpha) + c7`. Reads alpha from memory at `alpha_addr` and `alpha_addr + 1`.                                          |
```

Comment:
I think `tmp0` and `tmp1` are implementation details right? We could document it just as a standard horner eval, similar to how [wikipedia writes it](https://en.wikipedia.org/wiki/Horner%27s_method) - i.e. `c7 + alpha * (c6 + alpha * (...))`.

Reply:
Good point, removed and updated